### ISSUE TITLE
matmul bug on gfx803

### ISSUE THREAD

**Issue by Djip007** [ROCmSoftwarePlatform/tensorflow-upstream#485]
Using rocm 2.4 with RX-560X and tensorflow-rocm==1.13.3 with Python 3.7 (install on venv with pip on Fedora 30)... (yes I known it is not suported...)
matmul may have bad compute

```
import tensorflow as tf
import numpy as np
tf.enable_eager_execution()
# Define constant tensors
a = tf.constant([[2., 1.], [1., 0.]], dtype=tf.float32)
b = tf.constant([[3., 0.], [5., 1.]], dtype=tf.float32)
product = tf.matmul(a, b)
print(product)
#>>>
[[11.  1.]
 [ 3.  0.]]
```
OK
```
import tensorflow as tf
import numpy as np
tf.enable_eager_execution()
# Define constant tensors
a = tf.constant([[2., 1.], [1., 0.]], dtype=tf.float32)
b = np.array([[3., 0.], [5., 1.]], dtype=np.float32)
product = tf.matmul(a, b)
print(product)
#>>>
[[6. 0.]
 [3. 0.]]
```
KO  => a simple a*b is compute.

I have the same result (comment the line "tf.enable_eager_execution()" it is on by default...) on docker: using 
```
drun rocm/tensorflow:rocm2.4-tf2.0-alpha0-config-v2
```

**Comment by sunway513** [comment]
Hi @Djip007 , I'm able to reproduce your observations, and the calculation error is only reproducible on GFX803 target, and potentially caused by bad code gen. 
I've created an internal ticket to track this issue.

**Comment by maweil** [comment]
FYI: The bug also occurs when using an Radeon RX 480 and the official docker containers `rocm/tensorflow:latest` (which is tf-1.13.1) and  `rocm/tensorflow:rocm2.5-tf2.0-beta1-config-v2`. I'm also using Fedora 30 with Kernel version `5.1.15-300`.

For me, the example with tensorflow constants computes the wrong result, too:
```python
import tensorflow as tf
import numpy as np
#tf.enable_eager_execution()
# Define constant tensors
a = tf.constant([[2., 1.], [1., 0.]], dtype=tf.float32)
b = tf.constant([[3., 0.], [5., 1.]], dtype=tf.float32)
product = tf.matmul(a, b)
print("Result with ROCm TensorFlow {} in Docker Container\n {}".format(tf.__version__, product))
```
Result:
```
Result with ROCm TensorFlow 2.0.0-beta1 in Docker Container
 [[6. 0.]
 [3. 0.]]
```

**Comment by abderrahim** [comment]
It seems this is fixed in rocm 2.6 (though I had to rebuild tensorflow from source because of #559)

**Comment by Djip007** [comment]
sorry for late... test with:
drun rocm/tensorflow:rocm2.6-tf1.14-python3
drun rocm/tensorflow:rocm2.7-tf2.0-beta1-config-v2 
all work

(test with docker-ce on fedora-30!!!)
